src/documents/bulk_edit.py

```diff
@@ -146,3 +157,127 @@ def set_permissions(doc_ids, set_permissions, owner=None, merge=False):
     bulk_update_documents.delay(document_ids=affected_docs)
 
     return "OK"
+
+
+def rotate(doc_ids: list[int], degrees: int):
+    logger.info(
+        f"Attempting to rotate {len(doc_ids)} documents by {degrees} degrees.",
+    )
+    qs = Document.objects.filter(id__in=doc_ids)
+    affected_docs = []
+    import pikepdf
+
+    for doc in qs:
+        try:
+            with pikepdf.open(doc.source_path, allow_overwriting_input=True) as pdf:
+                for page in pdf.pages:
+                    page.rotate(degrees, relative=True)
+                pdf.save()
+                doc.checksum = hashlib.md5(doc.source_file.read()).hexdigest()
+                doc.save()
+                update_document_archive_file.delay(
```

Comment:
Although unlikely, I'm concerned about a race condition between the 2 async tasks here.

I think a [celery chord or chain or just group](https://docs.celeryq.dev/en/stable/userguide/canvas.html#the-primitives) would prevent that, so these first task(s) are done before the bulk update can happen

Reply:
Took a stab at this, let me know if thats what you had in mind (or not)